tests/update/patch_before_migration.pm

```diff
@@ -73,6 +79,56 @@ sub patching_sle {
     set_var("SYSTEM_PATCHED", 1);
 }
 
+sub install_package {
+    my @pk_list = split(/,/, get_var('PACKAGES'));
+    for my $pk (@pk_list) {
+        # removed package if starting with -
+        if ($pk =~ /^-/) {
+            zypper_call "rm -t package $pk";
+        }
+        else {
+            zypper_call "in -t package $pk";
+        }
+    }
+}
+
+sub install_patterns {
+
+    my $pcm = 0;
+    my @pt_list;
+    my @pt_list_un;
+    my @pt_list_in;
+
+    # install all patterns from product.
+    if (check_var('PATTERNS', 'all')) {
+        @pt_list_un = split(/ /, script_output("zypper pt -u | grep '^ ' | awk '{print \$2}' | sort -u | xargs"));
+        $pcm = grep /Amazon-Web-Services|Google-Cloud-Platform|Microsoft-Azure/, @pt_list_in;
```

Comment:
The pt_list_in is not valued yet at this point. And seems line 118 is really wanted?

Reply:
Updated, move pt_list_in definition to the beginning.

Reply:
Duplicates between line 106 and 118? Why check $pcm twice?

Reply:
yeah, this line is Duplicated

Reply:
From the snippet, I suspect that we will always install `Amazon-Web-Services*` patterns and other two would never be installed and tested.